dashboard/src/actions/overviewActions.js

```diff
@@ -523,3 +531,42 @@ const nonEssentialKeys = [
 
 const isServerInternal = (string) =>
   nonEssentialKeys.some((e) => string.match(e));
+
+/**
+ * Function to update metadata
+ * @function
+ * @param {String} path - nested key to update
+ * @param {Object} obj - nested object
+ * @param {String} obj - new value to be updated in the object
+ * @return {Object} - updated object with new value
+ */
+
+const setValueFromPath = (path, obj, value) => {
+  const [head, ...rest] = path.split(".");
+
+  return {
+    ...obj,
+    [head]: rest.length
+      ? setValueFromPath(rest.join("."), obj[head], value)
+      : value,
+  };
+};
```

Comment:
This is very clever!  However, the fact that we split the path on entry and then rejoin it in the subroutine call is unsatisfying.  How about something like this:
```javascript
const setValueFromPath = (path, obj, value) => {
  const recurse = (plist, o, v) => {
    const [head, ...rest] = plist;
    return {...o, [head]: rest.length ? recurse(rest, o[head], v) : v};
  };

  return recurse(path.split("."), obj, value);
};
```
That is, we split the path once, on the initial call to the recursive function, and after that we just manipulate the array.